packages/libs/error-stack/Cargo.toml

```
@@ -17,21 +17,25 @@ tracing-error = { version = "0.2.0", optional = true }
 once_cell = { version = "1.10.0", optional = true }
 pin-project = { version = "1.0.10", optional = true }
 futures-core = { version = "0.3.21", optional = true, default-features = false }
+smallvec = { version = "1.9.0", optional = true }
```

I think we want the `union` feature for an even smaller small vec?
```suggestion
smallvec = { version = "1.9.0", optional = true, default-features = false, features = ["union"] }
```

---

I thought about this, but I wanted to keep this at the discretion of the user. Whenever the user uses small, I assume they use `smallvec` themselves, and if they wanted, they could toggle `union`. Otherwise, we would add `union` for them, even if they use it in a different use-case, which might be surprising if that makes sense. (at least that's what I understood from [feature unifcation](https://doc.rust-lang.org/cargo/reference/features.html#feature-unification). Maybe creating a note in the docs would be a good idea?

---

That's a good point. Could you add that to the documentation next to the feature flags at the top-level module docs?
I think we want to set `default-features = false` (even if it has no effect currently) anyway? (Should have been done for the other dependencies as well, which was probably an oversight, feel free to drive-by these).